### ISSUE TITLE
Google Oauth returns an error after access token expires

### ISSUE THREAD

**Issue by colbyfayock** [nasa/cumulus#829]
The application seems to be throwing an error once the access token expires, as there's no refresh token except after the first authorization.

To get around this for login purposes, you can revoke the access in the Google Account security management.

As far as reproducing this:
 - Log in via Google Oauth
 - Hit log out
 - Immediately try to log back in with same account
 - You should hit an error, which may be different depending on version.

1.11.2 looks like:
```
{
statusCode: 401,
error: "Unauthorized",
message: "The record has validation errors"
}
```

I'm following up this issue with a PR

**Comment by markdboyd** [comment]
Interestingly, I'm hitting the 401 error on my first attempt to log in to the dashboard via Google. Perhaps I have something misconfigured, so let me dive into this a bit deeper and see what I find.

**Comment by markdboyd** [comment]
I spent a bit more time on this and confirmed the behavior you are experiencing. It looks like [the internet confirms it as well](https://stackoverflow.com/a/10857806).

However, that SO post suggests another possible solution: adding a parameter of `prompt` with a value of `consent` to the authorization URL. This will cause authorization with Google to always return a refreshToken. The downside is that it will also ask you to re-authorize the app every time you login. I [tested this solution locally](https://github.com/nasa/cumulus/compare/test-google-refreshToken?expand=1#diff-a72bc3ef5e79854ad10f68f8eec8baedR43) and it seemed to resolve the issue as well.

There might be some minor annoyance to authorizing the app every time you login via Google, but just to be clear it **is not** asking you to re-authenticate (re-enter your username/password) every time. You just have to re-click your account name to authorize it for login:

<img width="1440" alt="screen shot 2019-02-21 at 9 48 21 pm" src="https://user-images.githubusercontent.com/1001694/53216456-9f3eed00-3622-11e9-827d-7b0e3253181f.png">

I'm also somewhat hesitant about removing the `refreshToken` as a required field in the schema, because for Earthdata login it is necessary to support automatic session refreshes. So If we removed it as a required field and had some future regression where refreshTokens were not returned for EDL, then session refreshes would ultimately fail, but we wouldn't get the same immediate feedback from the schema failure on login.

I'm inclined towards the `prompt=consent` as the solution here. What do you think?

cc @yjpa7145 - Do you have any thoughts on the preferred solution here?

**Comment by colbyfayock** [comment]
yeah, I veered towards the schema solution mostly because I saw it as a bit of a nuisance for the auth prompt every time from a UX perspective. that said, you make a fair argument, especially considering Earthdata is the larger use case here and i wouldn't consider getting prompted again a dealbreaker, especially given that it errors at the moment

another consideration is how oauth providers other than google handle a similar situation, though I'm generally not familiar with the technicalities of the standard. perhaps that could help make an argument one way or another. i can try to dig a little on that if it's a worthwhile consideration

**Comment by yjpa7145** [comment]
> I'm also somewhat hesitant about removing the `refreshToken` as a required field in the schema, because for Earthdata login it is necessary to support automatic session refreshes. So If we removed it as a required field and had some future regression where refreshTokens were not returned for EDL, then session refreshes would ultimately fail, but we wouldn't get the same immediate feedback from the schema failure on login.

If there are going to be cases where an access token does not have a refresh token, I don't have a problem with making that an optional field.  If the concern is that Earthdata Login could change and not send a refresh token back, we could always add an explicit check here: https://github.com/nasa/cumulus/blob/db55cacd635e7c90e302d0d81c379643a9fd85e5/packages/api/lib/EarthdataLogin.js#L143-L144

**Comment by yjpa7145** [comment]
@colbyfayock Are you using the Cumulus Dashboard when this is happening?

**Comment by colbyfayock** [comment]
@yjpa7145 yup - cumulus dashboard

**Comment by markdboyd** [comment]
OK. I'm fine to go with dropping refreshToken as a required field and adding more strict checks for Earthdata login if necessary. 

@colbyfayock - Our integration tests still don't run correctly on forked PRs, which blocks the PR from getting merged. So I'm going to copy your changes into another branch I create, submit a PR from that branch, and close your PR. I'm sorry, I know this is a non-ideal workflow.

**Comment by colbyfayock** [comment]
@markdboyd no worries, I saw that part in the contrib guide so was expecting it :) thanks for the heads up

**Comment by colbyfayock** [comment]
thanks @markdboyd - any idea when i can expect this in a release available via npm?

**Comment by markdboyd** [comment]
We don't have a release date for the next release yet. 

What is your use case for Cumulus? Is this a major blocker for you?

**Comment by colbyfayock** [comment]
We're using it for a data processing pipeline. Not a major blocker at the moment.

**Comment by laurenfrederick** [comment]
@colbyfayock This has been released in 1.11.3

**Comment by colbyfayock** [comment]
@laurenfrederick awesome, thanks for the heads up 🙌
